src/screens/Profile/Notification.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { View, StyleSheet, AsyncStorage } from 'react-native';
+import { Card, Button, Paragraph } from 'react-native-paper';
+import auth from '@react-native-firebase/auth';
+import Swipeable from 'react-native-gesture-handler/Swipeable';
+import PropTypes from 'prop-types';
+
+export default class Notification extends React.Component {
+  constructor(props) {
+    super(props);
+    const { viewed } = this.props;
+    this.state = { viewed, visible: true };
+    this.onClearPress = this.onClearPress.bind(this);
+    this.onViewPress = this.onViewPress.bind(this);
+  }
+
+  componentDidMount() {
+    this.unsubscribeFromAuth = auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({ currentUserID: user.uid });
+      }
+    });
+  }
+
+  async onViewPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then((value) => {
+      const notificationArray = JSON.parse(value);
+      const notif = notificationArray.find((el) => el.id === notificationID);
+      if (notif.viewed === false) {
+        const updatedArray = notificationArray.filter((val) => val.id !== notificationID);
+        notif.viewed = true;
+        updatedArray.push(notif);
+        AsyncStorage.setItem(currentUserID, JSON.stringify(updatedArray));
+        this.setState({ viewed: true });
+      }
+    });
+  }
+
+  async onClearPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then(async (value) => {
+      if (value) {
+        const notificationArray = JSON.parse(value);
+        const updatedArray = notificationArray.filter((val) => val.id !== notificationID);
+        await AsyncStorage.setItem(currentUserID, JSON.stringify(updatedArray));
+        this.setState({ visible: false });
+      }
+    });
+  }
+
+  render() {
+    const { message, type, replies } = this.props;
+    const { viewed, visible } = this.state;
+    const active = viewed ? notificationStyles.inactiveNotification
+      : notificationStyles.activeNotification;
+    const replyText = ` has received ${replies} replies. Click to go to post`;
+    const approvedText = ' has been approved. Click to go to post';
+    let text;
+    if (type === 'reply') {
+      text = replyText;
+    } else if (type === 'approve') {
+      text = approvedText;
+    }
```

Comment:
You can use an object to switch between the text. That way we can easily add new strings. This object can go in the constructor.

```
this.notificationDescriptions = {
  reply: `Your post ${message} has recieved ${replies} replies. Click to go to post.`,
  approve: `Your post ${message} has been approved. Click to go to post.`,
};
```

You can also rename `message` to `postTitle` so it's more descriptive.